manifests/agent.pp

```diff
@@ -0,0 +1,51 @@
+class ss_nessus_agent::agent inherits ::ss_nessus_agent {
+
+  if $ss_nessus_agent::http_proxy {
+      $proxy_environment = ["http_proxy=${ss_nessus_agent::http_proxy}", "https_proxy=${ss_nessus_agent::http_proxy}"]
+  } else {
+      $proxy_environment = []
+  }
+
+  exec { 'download':
+    command     => "curl -s -f ${ss_nessus_agent::download_url} -o /usr/src/nessusagent.deb",
+    path        => '/usr/bin:/usr/sbin:/bin',
+    onlyif      => "test ! -f /usr/src/nessusagent.deb",
+    environment => $proxy_environment,
+  }
+  -> package { 'nessusagent':
+    ensure   => installed,
+    provider => dpkg,
+    source   => '/usr/src/nessusagent.deb',
+  }
+
+  # The setup and configuration utilises the custom facts
+  if $params::nessus_agent_enabled == true {
+    if !$facts['ss_nessus_agent_connected'] {
+      exec { 'ss_nessus_agent_install_agent':
```

Comment:
You are installing Nessus Agent twice in this file.
Initially after the download it is being installed.

```
-> package { 'nessusagent':
    ensure   => installed,
    provider => dpkg,
    source   => '/usr/src/nessusagent.deb',
  }
```

And then here it is being re-installed.